src/components/PlanetsList.test.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import {
+  render, cleanup, waitForElementToBeRemoved,
+} from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+
+import Provider from '../contexts/Provider';
+import Table from './Table';
+import { SW_BASE_API } from '../services/SWapi';
+import mockFetchPromise, { planets } from '../__mocks__/apiMock';
+
+const testTableRender = async () => {
+  jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise);
+
+  const { queryByTestId, queryByText, getByRole } = render(
+    <Provider>
+      <Table />
+    </Provider>,
+  );
+
+  expect(queryByText(/Loading/i)).toBeInTheDocument();
+
+  await waitForElementToBeRemoved(() => queryByText(/Loadin/i));
+
+  expect(global.fetch).toHaveBeenCalledWith(`${SW_BASE_API}/planets`);
+
+  expect(queryByText(/Loading/i)).toBeNull();
+
+  return {
+    queryByText,
+    getByRole,
+    queryByTestId,
+  };
+};
+
+afterEach(cleanup);
+
+describe('PlanetsList tests', () => {
+  test('if exist a table', async () => {
+    const { getByRole } = await testTableRender();
+
+    expect(getByRole('table')).toBeInTheDocument();
+  });
+  test('if exist a table thead', async () => {
+    const { queryByTestId } = await testTableRender();
+
+    Object.keys(planets[0]).forEach((key) => {
+      expect(queryByTestId(`th-${key}`)).toBeInTheDocument();
+      expect(queryByTestId(`th-${key}`).tagName).toBe('TH');
+    });
+  });
+  test('if exist a table tbody', async () => {
+    const { queryByText } = await testTableRender();
+
+    planets.map((planet) => Object.values(planet).forEach((value, index) => {
+      if (index === 9) {
+        expect(queryByText(new RegExp(value[0] || value, 'i'))).toBeInTheDocument();
+      } else {
+        expect(queryByText(value)).toBeInTheDocument();
```

Comment:
Não há necessidade de um `else` aqui. Pode colocar um `if` sem um else, porque ele vai se enquadrar nessa determinação apenas se o if for igual a 9.
```suggestion
        if (index === 9) {
          expect(queryByText(new RegExp(value[0] || value, 'i'))).toBeInTheDocument();
        }
        expect(queryByText(value)).toBeInTheDocument();
```

Reply:
Esse else eu acho que ele é necessário porque como o if nao tem uma condicao de parada se eu tirar o else, simplesmente qndo for passar pelo if ele tbm vai passar fora do if(justamente pq n tem um break ou return dentro do if). Mas a ideia do ternario é muito valida sim, vou mudar.